### ISSUE TITLE
Rule S3267: FP, issue reported for DataColumnCollection

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4996]
### Description

An issue is raised when iterating `DataColumnCollection` which doesn't implement `IEnumerable<T>`

### Repro steps

```
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;

namespace Tests.Diagnostics
{
    public class Loops
    {
        public void M(DataTable rawSheetData)
        {
            var row = rawSheetData.Rows[0];
            var data = new Dictionary<string, string>();
            foreach (DataColumn column in rawSheetData.Columns) // Noncompliant - FP
            {
                var datum = Convert.ToString(row[column.ColumnName]);
                data.Add(column.ColumnName, datum);
            }
        }
    }
}

```
### Expected behavior

We should not raise in this context since Select cannot be used without a cast.

### Actual behavior

An issue is raised.

### Known workarounds

Use `Cast<T>()` and then `Select()`
